pkg/datamover/repository_server.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"context"
+	"github.com/pkg/errors"
+
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kopia/snapshot"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+// Check that RepositoryServer implements DataMover interface
+var _ DataMover = (*RepositoryServer)(nil)
+
+type RepositoryServer struct {
```

Comment:
This can also be local.

Reply:
Similarly, as https://github.com/kanisterio/kanister/pull/1951/files#r1227623986

Reply:
Done !